src/wallet.js

```diff
@@ -690,24 +696,49 @@ const wallet = {
   },
 
   /**
-   * Check if unspentTx is locked or can be used
+   * Check if unspentTx is locked or can be used (only the part of the method that can be syncronous)
    *
    * @param {Object} unspentTx (needs to have decoded.timelock key)
+   * @param {number} blockHeight If unspentTx is a block reward, it's the height of the block. It's optional, for the case of transaction.
+   * @param {number} currentHeight Height of the last block of the network. It's optional, only in case we want to validate the block reward lock (not used in balance method)
    *
    * @return {boolean}
    *
    * @memberof Wallet
    * @inner
    */
-  canUseUnspentTx(unspentTx) {
+  canUseUnspentTxSync(unspentTx, blockHeight, currentHeight) {
     if (unspentTx.decoded.timelock) {
       let currentTimestamp = dateFormatter.dateToTimestamp(new Date());
       return currentTimestamp > unspentTx.decoded.timelock;
+    } else if (blockHeight) {
+      return (currentHeight - blockHeight) >= this.getRewardLockConstant();
     } else {
       return true;
     }
   },
 
+  /**
+   * Check if unspentTx is locked or can be used
+   *
+   * @param {Object} unspentTx (needs to have decoded.timelock key)
+   * @param {number} blockHeight If unspentTx is a block reward, it's the height of the block. It's optional, for the case of transaction.
+   * @param {number} currentHeight Height of the last block of the network. If not passed, we get it from the API.
+   *
+   * @return {Promise} Promise resolving a boolean if the utxo can be spent
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  async canUseUnspentTx(unspentTx, blockHeight, currentHeight) {
+    if (currentHeight === undefined && blockHeight) {
+      const response = await walletApi.getMiningInfoRequest();
```

Comment:
You can check if current height is enough. If it is, then you can return without running the `getMiningInfoRequest`, otherwise you do the request.

Reply:
I only execute the API call if it's not passed as parameter.